File: ProyectoMysql/server/DataLayer/EntListaDB.py
from EntityLayer.EntListaEntity import EntListaEntity, EntListaItemModel
from EntityLayer.MerListaEntity import MerListaSaveModel
from Utilidades.Entidades.ResponseAPI import ResponseAPI
from Utilidades.Conexion.ErrorData import ErrorData
from Utilidades.Conexion.configMysql import DBProcedure, Restore,CloseConnection
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum


class EntListaDB:


    def RegistrarDB(Ent: MerListaSaveModel):
        try:
   
            store_mapping = {
                ProcessActionEnum.Update: "sp_MerLista_Actualizar",
                ProcessActionEnum.Add: "sp_MerLista_Registrar",
            }
            Store = store_mapping.get(Ent.Action, "sp_MerLista_Registrar")
            args = []
            args.append(Ent.ListaId)
            args.append(Ent.CampoId)
            args.append(Ent.Codigo)
            args.append(Ent.Nombre)
            args.append(Ent.Descripcion)
            args.append(Ent.CodUsuario)
            args.append(Ent.FechaRegistro)
            args.append(Ent.EstadoRegistro)
            args.append(Ent.CodigoTabla)
            Ent.ListaId = DBProcedure().DBProcedureInsertUpdate(
                Store, args, "v_ListaId"
            )

            return Ent
        except Exception as e:
            print(e)
            Restore()
            
    def BuscarItem(Codigo: str, Nombre : str ):
        try:
            args = (Codigo,Nombre)
            resulset = DBProcedure().DBProcedureConsult("sp_EnListaBuscarItem", args)
            list = [EntListaItemModel.Cargar(row) for row in resulset]
            return list
        except Exception as e:
            print(e)


    def ObtenerItem(ListaId: int):
        try:
            args = (ListaId,)
            resulset = DBProcedure().DBProcedureConsult("sp_EnListaObtenerItem", args)
            list = [EntListaItemModel.Cargar(row) for row in resulset]
            return list
        except Exception as e:
            print(e)


    def ObtenerItems(Codigo: str):
        try:
            args = (Codigo,)
            resulset = DBProcedure().DBProcedureConsult("sp_EnListaObtenerItems", args)
            list = [EntListaItemModel.Cargar(row) for row in resulset]
            return list
        except Exception as e:
            print(e)

File: Facturacion/FactCore/FactCore.DataLayer/EntListaDB.cs
﻿using FactCore.Common;
using FactCore.EntityLayer;
using Framework;
using System.Data;

namespace FactCore.DataLayer

{
    public class EntListaDB : BaseDataEntity
    {

        public virtual bool Registrar(EntListaEntity Ent)
        {
            StartHelper(true);
            try
            {
                RegistrarDB(Ent);
            }
            catch (Exception ex)
            {
                Helper.CancelTransaction();
                throw ex;
            }

            Helper.Close();
            return true;
        }

        private bool RegistrarDB(EntListaEntity Ent)
        {
            if (Ent.LogicalState == LogicalState.Added || Ent.LogicalState == LogicalState.Updated)
            {
                String storedName = "sp_MerLista_Actualizar";
                if (Ent.LogicalState == LogicalState.Added) storedName = "sp_MerLista_Registrar";
                DbDatabase.GetStoredProcCommand(storedName);
                DbDatabase.SetTransaction(Helper.DbTransaction);

                DbDatabase.AddParameter(MyUtils.GetOutputDirection(true), "v_ListaId", DbType.Int32, 4, false, 0, 0, Ent.ListaId);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_CampoId", DbType.Int32, 4, false, 0, 0, Ent.CampoId);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Codigo", DbType.String, 100, false, 0, 0, Ent.Codigo);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Nombre", DbType.String, 100, false, 0, 0, Ent.Nombre);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Descripcion", DbType.String, 100, false, 0, 0, Ent.Descripcion);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_CodUsuario", DbType.String, 20, false, 0, 0, Ent.CodUsuario);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_FechaRegistro", DbType.DateTime, 12, false, 0, 0, Ent.FechaRegistro);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_EstadoRegistro", DbType.Boolean, 2, false, 0, 0, Ent.EstadoRegistro);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_CodigoTabla", DbType.String, 100, false, 0, 0, Ent.CodigoTabla);
                int returnValue = DbDatabase.ExecuteNonQuery();
                if (Ent.LogicalState == LogicalState.Added)
                {
                    if (Ent.ListaId <= 0) Ent.ListaId = (Int32)DbDatabase.GetParameterValue("v_ListaId");
                    Ent.OnLogicalAdded();
                }
                else
                {
                    if (returnValue <= 0) throw new Exception("ErrorDB.UpdateEntity");
                    Ent.OnLogicalUpdate();
                }
            }


            return true;
        }
      
        public virtual List<EntListaEntity> BuscarItem(String Codigo, String Nombre)
        {
            try
            {
                StartHelper(false);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Codigo", DbType.String, 50, false, 0, 0, Codigo);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Nombre", DbType.String, 50, false, 0, 0, Nombre);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(CommandType.StoredProcedure, "sp_EnListaBuscarItem");
                FillSchemeTable(dr);
                List<EntListaEntity> EntityList = new List<EntListaEntity>();
                while (dr.Read())
                {
                    EntListaEntity entity = new EntListaEntity();
                    if (FillFrom(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public virtual List<EntListaEntity> ObtenerItem(Int32 MercaderiaId)
        {
            try
            {
                StartHelper(false);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_ListaId", DbType.Int32, 4, false, 0, 0, MercaderiaId);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(CommandType.StoredProcedure, "sp_EnListaObtenerItem");
                FillSchemeTable(dr);
                List<EntListaEntity> EntityList = new List<EntListaEntity>();
                while (dr.Read())
                {
                    EntListaEntity entity = new EntListaEntity();
                    if (FillFrom(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public virtual List<EntListaEntity> ObtenerItems(String Codigo)
        {
            try
            {
                StartHelper(false);
                DbDatabase.AddParameter(MyUtils.GetOutputDirection(false), "v_Codigo", DbType.String, 50, false, 0, 0, Codigo);
                IDataReader dr = (IDataReader)DbDatabase.ExecuteReader(CommandType.StoredProcedure, "sp_EnListaObtenerItems");
                FillSchemeTable(dr);
                List<EntListaEntity> EntityList = new List<EntListaEntity>();
                while (dr.Read())
                {
                    EntListaEntity entity = new EntListaEntity();
                    if (FillFrom(dr, entity)) EntityList.Add(entity);
                    entity.OnLogicalLoaded();
                }

                Helper.Close(dr);
                return EntityList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
    }
}
